x/authenticator/types/codec.go

```diff
@@ -6,10 +6,18 @@ import (
 	"github.com/cosmos/cosmos-sdk/types/msgservice"
 )
 
+type AuthenticatorTxOptions interface {
+	GetSelectedAuthenticators() []int32
+}
+
 func RegisterCodec(cdc *codec.LegacyAmino) {
 }
 
 func RegisterInterfaces(registry cdctypes.InterfaceRegistry) {
+	registry.RegisterImplementations(
+		(*AuthenticatorTxOptions)(nil),
```

Comment:
Is there a way to do this without declaring an interface? why doesn't it work without?